Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp

```diff
@@ -182,6 +182,45 @@ static std::optional<std::pair<uint32_t, uint32_t>> findCrtc(int fd)
 }
 #endif
 
+using DrmNode = UnixFileDescriptor;
+using CrtcInfo = std::pair<uint32_t, uint32_t>;
+#if PLATFORM(GTK) || (PLATFORM(WPE) && ENABLE(WPE_PLATFORM))
+static std::optional<std::pair<DrmNode, CrtcInfo>> findDrmNodeWithCrtc(PlatformMonitor* monitor = nullptr)
+#else
+static std::optional<std::pair<DrmNode, CrtcInfo>> findDrmNodeWithCrtc()
+#endif
+{
+    constexpr int maxDrmDevices = 256;
+    drmDevicePtr devices[maxDrmDevices] = { 0 };
+    const int devicesNum = drmGetDevices(devices, maxDrmDevices);
+    if (devicesNum <= 0)
+        return { };
+    for (int i = 0; i < devicesNum; i++) {
+        if (devices[i]->available_nodes & (1 << DRM_NODE_PRIMARY)) {
```

Comment:
I would continue if it doesn't have a primary node and we save an indentation level for the rest of the for body.